### DISCUSSION TITLE
Feature request: Support double click, shift click on `TransferList`

### DISCUSSION THREAD

**Discussion by HaveSpacesuit** [iTwin/iTwinUI#2018] [Ideas]
I'm replacing our custom build transfer list with the iTwin [TransferList](https://itwinui.bentley.com/docs/transferlist). A couple features are missing in the iTwin implementation, and it would be great if we could see them added:

* Double click a list item -> transfers to the opposite list
* Shift+click within a list -> selects all list items between last selected and where Shift+click was applied (same as the Shift+click in the table)

**Comment by mayank99** [comment]
I don't think we would add the double-click functionality within iTwinUI, because it feels too arbitrary (you are still free to add it in your product if you explain it well to your users). However, shift+click makes sense to add, as it is fairly common and more intuitive.

Since we already have shift+click implemented in `Table`, maybe it would also make sense to abstract that logic into a common helper function that can be used in both places? Might be difficult though, as it relies on implementation details of the `Table` component. The code can be found here:

https://github.com/iTwin/iTwinUI/blob/d1af8d045f7117476abc06467d98b314b92a88c5/packages/itwinui-react/src/core/Table/actionHandlers/selectHandler.ts#L123

That being said, we don't currently have the bandwidth to work on this. Maybe you'd like to [contribute](https://github.com/iTwin/iTwinUI/blob/main/CONTRIBUTING.md) this feature?
